src/librustc_driver/lib.rs

```diff
@@ -159,28 +182,111 @@ pub use rustc_trans_utils::trans_crate::MetadataOnlyTransCrate as DefaultTransCr
 pub use rustc_trans::LlvmTransCrate as DefaultTransCrate;
 
 #[cfg(not(feature="llvm"))]
-mod rustc_trans {
-    use syntax_pos::symbol::Symbol;
-    use rustc::session::Session;
-    use rustc::session::config::PrintRequest;
+pub mod rustc_trans {
     pub use rustc_trans_utils::trans_crate::MetadataOnlyTransCrate as LlvmTransCrate;
-    pub use rustc_trans_utils::trans_crate::TranslatedCrate as CrateTranslation;
 
-    pub fn init(_sess: &Session) {}
     pub fn print_version() {}
     pub fn print_passes() {}
-    pub fn print(_req: PrintRequest, _sess: &Session) {}
-    pub fn target_features(_sess: &Session) -> Vec<Symbol> { vec![] }
-
-    pub mod back {
-        pub mod write {
-            pub const RELOC_MODEL_ARGS: [(&'static str, ()); 0] = [];
-            pub const CODE_GEN_MODEL_ARGS: [(&'static str, ()); 0] = [];
-            pub const TLS_MODEL_ARGS: [(&'static str, ()); 0] = [];
+}
+
+fn load_backend_from_dylib(sess: &Session, backend_name: &str) -> Box<TransCrate> {
+    use std::sync::mpsc;
+    use std::path::Path;
+    use syntax::symbol::Symbol;
+    use rustc::session::config::OutputFilenames;
+    use rustc::ty::TyCtxt;
+    use rustc::ty::maps::Providers;
+    use rustc::middle::cstore::MetadataLoader;
+    use rustc::dep_graph::DepGraph;
+    use rustc_metadata::dynamic_lib::DynamicLibrary;
+    /// This prevents the dylib from being unloaded when there is still a TransCrate open
+    struct ExternTransCrate {
```

Comment:
Just a protip, but you should probably never unload the dylib at all, once it is loaded. Because of https://github.com/nagisa/rust_libloading/issues/5.

Reply:
Done